examples/service_account/src/main.rs

```diff
@@ -0,0 +1,183 @@
+//! $ target/debug/service_account sub
+//! Published message #95491011619126
+//! message <95491011619126> 'Hello oh wonderful world' at 2016-09-21T20:04:47.040Z
+//! Published message #95491011620879
+//! message <95491011620879> 'Hello oh wonderful world' at 2016-09-21T20:04:49.086Z
+//! Published message #95491011622600
+//! message <95491011622600> 'Hello oh wonderful world' at 2016-09-21T20:04:51.132Z
+//! Published message #95491011624393
+//! message <95491011624393> 'Hello oh wonderful world' at 2016-09-21T20:04:53.187Z
+//! Published message #95491011626206
+//! message <95491011626206> 'Hello oh wonderful world' at 2016-09-21T20:04:55.233Z
+//!
+//! Copyright (c) 2016 Google, Inc. (Lewin Bormann <lbo@spheniscida.de>)
+//!
+extern crate base64;
+extern crate yup_oauth2 as oauth;
+extern crate google_pubsub1 as pubsub;
+extern crate hyper;
+
+use std::env;
+use std::time;
+use std::thread;
+
+use pubsub::{Topic, Subscription};
+
+// The prefixes are important!
+const SUBSCRIPTION_NAME: &'static str = "projects/sanguine-rhythm-105020/subscriptions/rust_authd_sub_1";
+const TOPIC_NAME: &'static str = "projects/sanguine-rhythm-105020/topics/topic-01";
+
+type PubsubMethods<'a> = pubsub::ProjectMethods<'a,
+                                                hyper::Client,
+                                                oauth::ServiceAccountAccess<hyper::Client>>;
+
+// Verifies that the topic TOPIC_NAME exists, or creates it.
+fn check_or_create_topic(methods: &PubsubMethods) -> Topic {
+    let result = methods.topics_get(TOPIC_NAME).doit();
+
+    if result.is_err() {
+        println!("Assuming topic doesn't exist; creating topic");
+        let topic = pubsub::Topic { name: Some(TOPIC_NAME.to_string()) };
+        let result = methods.topics_create(topic, TOPIC_NAME).doit().unwrap();
+        result.1
+    } else {
+        result.unwrap().1
+    }
+}
+
+fn check_or_create_subscription(methods: &PubsubMethods) -> Subscription {
+    // check if subscription exists
+    let result = methods.subscriptions_get(SUBSCRIPTION_NAME).doit();
+
+    if result.is_err() {
+        println!("Assuming subscription doesn't exist; creating subscription");
+        let sub = pubsub::Subscription {
+            topic: Some(TOPIC_NAME.to_string()),
+            ack_deadline_seconds: Some(30),
+            push_config: None,
+            name: Some(SUBSCRIPTION_NAME.to_string()),
+        };
+        let (_resp, sub) = methods.subscriptions_create(sub, SUBSCRIPTION_NAME).doit().unwrap();
+
+        sub
+    } else {
+        result.unwrap().1
+    }
+}
+
+fn ack_message(methods: &PubsubMethods, id: String) {
+    let request = pubsub::AcknowledgeRequest { ack_ids: Some(vec![id]) };
+    let result = methods.subscriptions_acknowledge(request, SUBSCRIPTION_NAME).doit();
+
+    match result {
+        Err(e) => {
+            // There's a JSON decode error happening, even on successful returns. :/
+            println!("Ack error: {:?}", e);
```

Comment:
Hmm, maybe a problem in the code generated by google-apis-rs ? Or already fixed thanks to the `empty-structs` PR recently submitted ?